src/UXClient/Components/LineChart/LineChart.ts

```diff
@@ -1286,11 +1286,13 @@ class LineChart extends TemporalXAxisComponent {
                 aEO.swimLane = 0;
             });
             // consolidate horizontal markers
-            const horizontalMarkers = [];
-            Object.values(this.chartOptions.swimLaneOptions).forEach((lane) => {
-                horizontalMarkers.push(...lane.horizontalMarkers);
-            });
-            this.chartOptions.swimLaneOptions = {0: {yAxisType: this.chartOptions.yAxisState, horizontalMarkers: horizontalMarkers}};
+            if (this.chartOptions.swimLaneOptions) {
+                const horizontalMarkers = [];
+                Object.values(this.chartOptions.swimLaneOptions).forEach((lane) => {
+                    horizontalMarkers.push(...lane.horizontalMarkers);
+                });
+                this.chartOptions.swimLaneOptions = {0: {yAxisType: this.chartOptions.yAxisState, horizontalMarkers: horizontalMarkers}};
```

Comment:
Please do add comments about the fix, the point about ES2015/2017 etc.

Any chance you can add a test case ?

Also, what is the else case of the **if (this.chartOptions.swimLaneOptions) {** scenario ?

Reply:
each swim lane can have its own set of horizontal markers defined in the this.chartOptions.swimLaneOptions object like 
this.chartOptions.swimLaneOptions = {
  0: { yAxisType: 'shared', horizontalMarkers: [10, 20] },
  1: { yAxisType: 'shared', horizontalMarkers: [30, 40] },
  2: { yAxisType: 'shared', horizontalMarkers: [50, 60] }
};

and we consolidate all them to a single swimlane 
this.chartOptions.swimLaneOptions = {0: {yAxisType: this.chartOptions.yAxisState, horizontalMarkers: horizontalMarkers}};

But if we have this.chartOptions.swimLaneOptions as null, then consolidating horizontal markers is not required.